packages/frontend/src/components/ProjectAccess/ProjectAccessRow.tsx

```diff
@@ -16,81 +21,94 @@ import {
+                                {user.firstName} {user.lastName}
                             </Text>
                         )}
-                        {email && (
+                        {user.email && (
                             <Badge color="gray" size="xs" radius="xs">
-                                {email}
+                                {user.email}
                             </Badge>
                         )}
                     </Stack>
                 </td>
                 <td>
                     <Group>
-                        {onUpdate ? (
-                            <NativeSelect
-                                id="user-role"
-                                data={Object.values(ProjectMemberRole).map(
-                                    (orgMemberRole) => ({
-                                        value: orgMemberRole,
-                                        label: orgMemberRole.replace('_', ' '),
-                                    }),
-                                )}
-                                onChange={(e) => {
-                                    const newRole = e.target
-                                        .value as ProjectMemberRole;
-                                    onUpdate(newRole);
-                                }}
-                                variant="filled"
-                                size="xs"
-                                value={role}
-                                sx={{ flex: 1 }}
-                                error={
-                                    relevantOrgRole
-                                        ? `This user inherits the organization role: ${relevantOrgRole}`
-                                        : undefined
-                                }
-                            />
-                        ) : (
-                            <Tooltip
-                                withinPortal
-                                withArrow
-                                label={roleTooltip ? roleTooltip : undefined}
-                            >
+                        <Tooltip
+                            withinPortal
+                            withArrow
+                            disabled={highestRole.type === 'project'}
+                            label={`This user inherits the ${capitalize(
+                                highestRole.type,
+                            )} role: ${
+                                ProjectMemberRoleLabels[highestRole.role]
+                            }`}
+                        >
+                            {projectRole?.role ? (
+                                <NativeSelect
+                                    id="user-role"
+                                    w="150px"
+                                    error={highestRole.type !== 'project'}
+                                    size="xs"
+                                    data={Object.values(ProjectMemberRole).map(
```

Comment:
I would set a `disabled` while the request is loading - or similar

Reply:
@joaoviana updated!